New package: genie-1141

ScrelliCopter opened the pull request:
https://github.com/bkaradzic/GENie
Please excuse the Git, it's required by the pre-build configure scripts to set the version reported by `--version`. Upstream has an irritating way of designating the current stable release so something like this has to be done anyway.
I did my best to make sure the package builds & cross-compiles on every architecture xbps-src supports.

ScrelliCopter commented:
Spoke too soon... native 32 bit builds should be fixed now.

Piraty commented on srcpkgs/genie/template:
```diff
@@ -0,0 +1,48 @@
+maintainer="a dinosaur <nick@a-dinosaur.com>"
+license="BSD-3-Clause"
+homepage="https://github.com/bkaradzic/GENie"
+_commit=44918162588e56512ddac6264b08ca6ba4e67468
+
+do_fetch() {
+	git clone "${homepage}.git" "${pkgname}-${version}"
+	cd "${pkgname}-${version}"
+	git reset --hard "${_commit}"
+}
```
grep for `githash` to see how to fetch a commit from a github repo

ScrelliCopter replied:
@Piraty I changed the convention to match what you were referring to, or did you also mean replacing the usage of Git with just getting the commit directly from GitHub?

The reason I'm using Git is because [these](https://github.com/bkaradzic/GENie/blob/44918162588e56512ddac6264b08ca6ba4e67468/scripts/release.lua#L18) [lines](https://github.com/bkaradzic/GENie/blob/44918162588e56512ddac6264b08ca6ba4e67468/scripts/release.lua#L21) use `git` to glean information about the version, which fails without an actual Git repository.

I could try patching these lines myself because they're just getting strings that I already know about in the template, and then the hostmakedepends=git could be omitted entirely in favour of just fetching the commit directly. Which would be preferred?

Piraty replied:
1. a small hint about git requirement would have made it more obvious
2. you can write `version.h` (as defined [here](https://github.com/bkaradzic/GENie/blob/44918162588e56512ddac6264b08ca6ba4e67468/scripts/release.lua#L18-L27)) on your own with the information available in the template, as well as maybe even skip running that release script entirely and mimic what does in `pre_configure()`. or just patch out the lines calling `git`, whatever you like

ScrelliCopter replied:
I did mention it in my PR comment, it should be more obvious now from the template (the git requirement is gone anyway).
Rewriting the whole script would be absurd when the issue can be fixed with patches & sed, so I've just opted for that.

Piraty commented on srcpkgs/genie/template:
```diff
@@ -0,0 +1,48 @@
+license="BSD-3-Clause"
+homepage="https://github.com/bkaradzic/GENie"
+
+do_fetch() {
+	git clone "${homepage}.git" "${pkgname}-${version}"
+	cd "${pkgname}-${version}"
+	git reset --hard "${_githash}"
+}
```
please `distfiles="https://github.com/bkaradzic/GENie/archive/${_githash}.tar.gz"` instead of `git clone`

ScrelliCopter replied:
Done.

Chocimier commented:
Could you ask them to tag releases?

ScrelliCopter commented:
Upstream is unlikely to want to use release tags while the (unmodified) build scripts require Git.

Piraty commented on srcpkgs/genie/patches/no-git.diff:
```diff
@@ -0,0 +1,23 @@
+diff --git a/scripts/release.lua b/scripts/release.lua
```
suggestion: write `version.h` directly from the template and simply drop its handling in `release.lua`, might make it less fragile

ScrelliCopter replied:
Reasonable suggestion @Piraty, it is done.

Piraty reviewed (changes requested):
i put some suggestion in-code.
Afte i looked into genie's build system, I also require you to put comments on the parts where you build for the host and why, and ideally guard that by testing `$CROSS_BUILD`

ScrelliCopter commented on srcpkgs/genie/template:
```diff
@@ -0,0 +1,50 @@
+		;;
+	*) # '-m64' only valid for x86_64, ppc64 and messes up 32-bit & ARM builds
+		# so remove it
+		vsed 's/-m64 //' -i build/gmake.linux/genie.make
```
Upstream may be receptive to a patch that only applies that on Windows, otherwise I'm not sure why it's there. Anyway, changes applied.

ScrelliCopter commented on srcpkgs/genie/template:
```diff
@@ -0,0 +1,50 @@
+	x86_64*|ppc64*) # no changes needed
+		;;
+	*) # same as before but fixes cross-builds
+		vsed 's/-m64//' -i scripts/genie.lua
```
Ditto.

ScrelliCopter commented:
@Piraty 
> Afte i looked into genie's build system, I also require you to put comments on the parts where you build for the host and why
I added a small explanation to the start of `do_configure` that should make the intention clear.
> and ideally guard that by testing `$CROSS_BUILD`
I put `CC="${CC_host:-$CC}"` in the make that runs on the host and switched to the `gnu-makefile` build_style to handle the actual build. Tested & seems to work for both native & cross builds. Should that handle every case?
